src/Silex/ArgumentResolver/ApplicationArgumentResolver.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Silex\ArgumentResolver;
+
+use Pimple\Container;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\ArgumentResolverInterface;
+
+class ApplicationArgumentResolver implements ArgumentResolverInterface
+{
+    private $pimple;
+
+    /**
+     * @param Container $pimple
+     */
+    public function __construct(Container $pimple)
```

Comment:
given that the resolver is named ApplicationArgumentResolver, should it be typehinted as Application instead ?